test/filesystem.cc

```diff
@@ -135,6 +137,70 @@ TEST(Filesystem, extension) {
 TEST(Filesystem, file_size) {
 }
 
+TEST(Filesystem, concurrent_folder_create_delete) {
+
+  const char* create_subdir = "test/a/b/c/d/e/f/g/h/i/j";
```

Comment:
Can we use `std::string` instead of C-style strings here?

Reply:
Nit: `create_subdir` is framed as a verb, I usually think of verbs as functions or similar

Reply:
The decision to use a const std::string vs const char *... it depends on a lot of factors.  I generally program in a manner that is "as simple as possible".  I don't need the features offered by std::string in this case, so I decided to use a const char * string literal.  However, I can sense this team is big on using C++ across the board.  

My philosophy/coding style is to best match the existing conventions within the team/project - so I'm happy to use std::string in these instances.

Fair point about the name, Anders.

Reply:
I'm inclined to agree with @ktatterso . For this specific use-case, i.e string constants, `const char*` (or maybe `const char[]` if we want to use size, which is compile time optimized)) is more light-weight than `std::string`. If the string needs to be modified or accepted as a function arg (even a const arg), then I'd prefer to use `std::string` (because of its implicit memory mgmt). Just my 2¢.

Reply:
To be clear, I think the issue here is around style/consistency of our type choices over performance concerns. The latter isn't really an issue, since this is a unit test. On the other hand, we seem more likely to use an STL string container over C-style strings in many places where we have string constants (there are, of course, exceptions to this).

Just pointing this out as a theme, I have no strong preferences about what we're doing here.